src/Mod/Path/PathScripts/PathAreaOp.py

```diff
@@ -61,68 +62,126 @@ class ObjectOp(PathOp.ObjectOp):
     Provides standard features including debugging properties AreaParams,
     PathParams and removalshape, all hidden.
     The main reason for existence is to implement the standard interface
-    to Path.Area so subclasses only have to provide the shapes for the
-    operations.'''
+    to Path.Area so subclasses only have to provide the shapes, returned
+    from areaOpShapes() method in the subclass operations.'''
+
+    def initOperation(self, obj):
+        '''initOperation(obj) ... Contains common instructions
+        for execution in __init__() method of operation class and
+        calls initAreaOp() in subclass.
+        Do not overwrite, overwrite initAreaOp(obj) instead.'''
+        PathLog.track()
+
+        self.initAreaOp(obj)
+
+    def _initOpAttributes(self, obj):
+        '''_initOpAttributes(obj) ... Set class attributes that must be set
+        at new operation instantiation and also upon document restoration.'''
+        self.initWithRotation = False
+        self._initAreaOpAttributes(obj)
 
     def opFeatures(self, obj):
         '''opFeatures(obj) ... returns the base features supported by all Path.Area based operations.
         The standard feature list is OR'ed with the return value of areaOpFeatures().
         Do not overwrite, implement areaOpFeatures(obj) instead.'''
         return PathOp.FeatureTool | PathOp.FeatureDepths | PathOp.FeatureStepDown | PathOp.FeatureHeights | PathOp.FeatureStartPoint | self.areaOpFeatures(obj) | PathOp.FeatureCoolant
 
-    def areaOpFeatures(self, obj):
-        '''areaOpFeatures(obj) ... overwrite to add operation specific features.
-        Can safely be overwritten by subclasses.'''
-        # pylint: disable=unused-argument
-        return 0
+    def opPropertyDefinitions(self):
+        # List of tuples for property definitions
+        defs = [
+            ("App::PropertyEnumeration", "EnableRotation", "Rotation",
+                QtCore.QT_TRANSLATE_NOOP("App::Property", "Enable rotation to gain access to pockets/areas not normal to Z axis.")),
+            ("App::PropertyString", "AreaParams", "Path", ""),
+            ("App::PropertyString", "PathParams", "Path", ""),
+            ("Part::PropertyPartShape", "removalshape", "Path", "")
+        ]
+
+        if self:
+            # self.editorModes is a list of (prop_name, mode) tuples to be processed later
+            self.editorModes.extend([
+                ('removalshape', 2),
+                ('AreaParams', 2),
+                ('PathParams', 2)
+            ])
```

Comment:
If you switch to a data driven model I would recommend using class constants for thesef strings.